src/types/line.js

```diff
@@ -39,17 +68,29 @@ function limitLineToArea(p1, p2, area) {
 }
 
 export default class LineAnnotation extends Element {
-  intersects(x, y, epsilon) {
+  intersects(x, y, epsilon = 0.001) {
+    // Adapted from https://stackoverflow.com/a/6853926/25507
+    const sqr = v => v * v;
+    const {x: x1, y: y1, x2, y2} = this;
+    const dx = x2 - x1;
+    const dy = y2 - y1;
+    const lenSq = sqr(dx) + sqr(dy);
+    const t = lenSq === 0 ? -1 : ((x - x1) * dx + (y - y1) * dy) / lenSq;
+    let xx, yy;
+    if (t < 0) {
+      xx = x1;
+      yy = y1;
+    } else if (t > 1) {
+      xx = x2;
+      yy = y2;
+    } else {
+      xx = x1 + t * dx;
+      yy = y1 + t * dy;
+    }
+    return (sqr(x - xx) + sqr(y - yy)) < epsilon;
+  }
     epsilon = epsilon || 0.001;
-    const me = this;
-    const p1 = {x: me.x, y: me.y};
-    const p2 = {x: me.x2, y: me.y2};
-    const dy = interpolateY(x, p1, p2);
-    const dx = interpolateX(y, p1, p2);
-    return (
-      (!isFinite(dy) || Math.abs(y - dy) < epsilon) &&
-			(!isFinite(dx) || Math.abs(x - dx) < epsilon)
-    );
+    return distanceBetweenPointAndLineSegment(x, y, this.x, this.y, this.x2, this.y2) < epsilon;
```

Comment:
I could not get the suggestion to match all the lines it should have affected. So these lines should be removed and the `distanceBetweenPointAndLineSegment` function as its inlined above

Reply:
I should have paid more attention; I apologize. Fixed.